src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let connectedParts = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] != null) {
+        connectedParts += this[i];
+      }
```

Comment:
Don't forget about undefined
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
        connectedParts += this[i];
      }
```